src/libPMacc/include/traits/GetUniqueTypeId.hpp

```diff
@@ -78,21 +85,9 @@ struct GetUniqueTypeId
      */
     static const ResultType uid(uint64_t maxValue = (uint64_t(1) << (sizeof (ResultType) * CHAR_BIT)) - uint64_t(1))
```

Comment:
I think we have missed a bug here.
I get ```warning: left shift count >= width of type``` which is true because you shift the 1 to the left by 64=8*8 bits.
The result is a 1 in the 65. bit. This is then subtracted by 1 to get the largest possible value. Theoretically this is correct, but practically this is undefined.
For unsigned numbers it should be more correct to use ```~uint64_t(0)``` or ```std::numeric_limits<ResultType>::max()``` to get the maximum.